impl/src/main/java/org/apache/myfaces/util/token/TokenGenerator.java

```diff
@@ -31,23 +34,46 @@ public class TokenGenerator
 {
     private final AtomicLong seed;
     
+    private static Logger log = Logger.getLogger(TokenGenerator.class.getName());
+
+    // TODO -- make a web parameter or it would be nice 
+    // to consolidate RANDOM_KEY_IN_CSRF_SESSION_TOKEN_SECURE_RANDOM_ALGORITM, 
+    // and RANDOM_KEY_IN_VIEW_STATE_SESSION_TOKEN_SECURE_RANDOM_ALGORITHM
+    private String[] supportedAlgorithmsList = {"SHA256DRBG","DRBG","SHA1PRNG"};
+
     public TokenGenerator()
     {
         seed = new AtomicLong(generateSeed());
     }
     
     private long generateSeed()
     {
-        SecureRandom rng;
-        try
+        SecureRandom rng = null;
```

Comment:
i assume generateSeed will be called again and again
so please cache rng - always re-lookup it causes exceptions and exceptions each time

Reply:
It's only called within the constructor. 

generateSeed is only called once per instance but I don't thin there should be harm in making it static and caching it.

Reply:
then we should rework this completely

we should look all allgorithm on startup (MyFacesConfig?) and only remember the algorithm to use

TokenGenerator/SessionIdGenerator should not try and try over and over

Reply:
ping me if you did it